algorithm/index.mjs

```diff
@@ -1,17 +1,487 @@
+
+  console.log(`Tipo de tasa de interés: ${inputs.interestRateType}`);
+  console.log(`Tasa efectiva del periodo: ${periodicalInterestRate * 100}%\n`);
+
+  console.log(`Tasa descuento Ks equivalente: ${descuentoKs * 100}%`);
+  console.log(`Tasa descuento WACC equivalente: ${descuentoWACC * 100}%`);
+
+  // Cronograma de pagos
+
+  let tipoPlazo = "";
+  let saldoInicial = leasingAmount;
+  let cuota = 0;
+  let intereses = 0;
+  let amortizacion = 0;
+  let saldoFinal = 0;
+  const depreciacion = -(sellingValue / periodos);
+  let ahorroTributario = 0;
+  let montoIGVPeriodico = 0;
+  let flujoBruto = 0;
+  let flujoIGV = 0;
+  let flujoNeto = 0;
+
+  let totalIntereses = 0;
+  let totalAmortizacion = 0;
+  let totalCostosPeriodicos = 0;
+  let totalSeguro = 0;
+
+  let vnaFlujoBruto = 0;
+  let vnaFlujoNeto = 0;
+
+  const flujosBrutos = [leasingAmount];
+  const flujosNetos = [leasingAmount];
+
+  for (let periodo = 1; periodo <= periodos; periodo++) {
+    console.log(`= = = = = Periodo ${periodo} = = = = =`);
+
+    intereses = -(periodicalInterestRate * saldoInicial);
+
+    if (periodo < periodos) {
+      const input = await inquirer.prompt({
+        name: "plazoDeGracia",
+        type: "list",
+        message: `Escoja el tipo de plazo de gracia: `,
+        choices: ["Total", "Parcial", "Sin plazo de gracia"],
+      });
+
+      tipoPlazo = input.plazoDeGracia;
+    } else {
+      tipoPlazo = "Sin plazo de gracia";
+    }
+
+    if (tipoPlazo === "Total") {
+      cuota = 0;
+      amortizacion = 0;
+      saldoFinal = saldoInicial - intereses;
+    } else if (tipoPlazo === "Parcial") {
+      cuota = intereses;
+      amortizacion = 0;
+      saldoFinal = saldoInicial + amortizacion;
+    } else {
+      cuota = -(
+        (Math.pow(1 + periodicalInterestRate, periodos - periodo + 1) * saldoInicial * periodicalInterestRate) /
+        (Math.pow(1 + periodicalInterestRate, periodos - periodo + 1) - 1)
+      );
+      amortizacion = cuota - intereses;
+      saldoFinal = saldoInicial + amortizacion;
+    }
+
+    console.log(`\nSaldo inicial: S/ ${roundMoney(saldoInicial)}`);
+    console.log(`Intereses: S/ ${roundMoney(intereses)}`);
+    console.log(`Cuota: S/ ${roundMoney(cuota)}`);
+    console.log(`Amortización: S/ ${roundMoney(amortizacion)}`);
+    console.log(`Monto de seguro contra todo riesgo: S/ ${roundMoney(insuranceAmount)}`);
+    console.log(`Costos periódicos: S/. ${roundMoney(periodicCosts)}`);
+
+    ahorroTributario = (intereses + insuranceAmount + periodicCosts + depreciacion) * IR;
+    montoIGVPeriodico = (cuota + insuranceAmount + periodicCosts) * IGV;
+    flujoBruto = cuota + insuranceAmount + periodicCosts;
+
+    if (inputs.buyingOption && periodo === periodos) {
+      montoIGVPeriodico = (cuota + insuranceAmount + periodicCosts + montoRecompra) * IGV;
+      flujoBruto = cuota + insuranceAmount + periodicCosts + montoRecompra;
+      console.log(`Monto de recompra: S/ ${roundMoney(montoRecompra)}`);
```

Comment:
You can use the `Intl.NumberFormat` API here as well.

Reply:
Done